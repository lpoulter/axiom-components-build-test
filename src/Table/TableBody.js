import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Base from '../Base/Base';
import classnames from 'classnames';

export default class TableBody extends Component {
  static propTypes = {
    /** TableRows */
    children: PropTypes.node,
    className: PropTypes.string,
  };

  render() {
    const {
      children,
      className,
      ...rest
    } = this.props;

    const classes = classnames('ax-table__body', className);

    return (
      <Base { ...rest } Component="tbody" className={ classes } >
        { children }
      </Base>
    );
  }
}
