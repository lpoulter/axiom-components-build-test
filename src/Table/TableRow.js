import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Base from '../Base/Base';

export default class TableRow extends Component {
  static propTypes = {
    /** Specifies whether or not the row should have borders */
    borderless: PropTypes.bool,
    /** TableCells */
    children: PropTypes.node,
    className: PropTypes.string,
    /** Apply hover styling */
    hover: PropTypes.bool,
    /** When pCardrovided, applies styling to indicate the TableRow is clickable */
    onClick: PropTypes.func,
  };

  render() {
    const {
      children,
      className,
      borderless,
      hover,
      onClick,
      ...rest
    } = this.props;

    const classes = classnames('ax-table__row', {
      'ax-table__row--hover': hover,
      'ax-table__row--clickable': onClick,
      'ax-table__row--borderless': borderless,
    }, className);

    return (
      <Base className={ classes } { ...rest } Component="tr" onClick={ onClick }>
        { children }
      </Base>
    );
  }
}
