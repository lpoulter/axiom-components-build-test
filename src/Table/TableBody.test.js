import React from 'react';
import renderer from 'react-test-renderer';
import TableBody from './TableBody';
import TableRow from './TableRow';

const getComponent = () =>
  renderer.create(
    <TableBody>
      <TableRow />
    </TableBody>
  );

describe('TableBody', () => {
  it('renders', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
