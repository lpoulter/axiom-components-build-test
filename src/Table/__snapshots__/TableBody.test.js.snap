// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableBody renders 1`] = `
<tbody
  className="ax-table__body"
>
  <tr
    className="ax-table__row"
  />
</tbody>
`;
